glib/src/gstring.rs

```diff
@@ -573,6 +573,63 @@ impl GString {
         Self(Inner::Native(None))
     }
     // rustdoc-stripper-ignore-next
+    /// Formats an [`Arguments`](std::fmt::Arguments) into a [`GString`].
+    ///
+    /// This function is the same as [`std::fmt::format`], except it returns a [`GString`]. The
+    /// [`Arguments`](std::fmt::Arguments) instance can be created with the
+    /// [`format_args!`](std::format_args) macro.
+    ///
+    /// Please note that using [`gformat!`] might be preferable.
+    pub fn format(args: fmt::Arguments) -> Self {
+        #[repr(transparent)]
+        struct GStringInline(ffi::GString);
+
+        impl fmt::Write for GStringInline {
+            #[inline]
+            fn write_str(&mut self, s: &str) -> fmt::Result {
+                unsafe {
+                    ffi::g_string_append_len(
+                        &mut self.0,
+                        s.as_ptr() as *const _,
+                        s.len().try_into().map_err(|_| fmt::Error)?,
+                    );
+                }
+                Ok(())
+            }
+            #[inline]
+            fn write_char(&mut self, c: char) -> fmt::Result {
+                unsafe {
+                    ffi::g_string_append_unichar(&mut self.0, c.into_glib());
+                }
+                Ok(())
+            }
+        }
+
+        impl Drop for GStringInline {
+            fn drop(&mut self) {
+                unsafe { ffi::g_free(self.0.str as *mut _) };
+            }
+        }
+
+        let mut s = GStringInline(ffi::GString {
```

Comment:
Why not go via `GStringBuilder`?

Reply:
Ah because you inline-allocate it here. Should maybe do the same for `GStringBuilder` though?

Reply:
`GStringBuilder` now uses `BoxedInline`